### PR TITLE
Update to 1.18.1

### PR DESCRIPTION
**Pull request by james090500**
Updates to 1.18

To Do
- [x] Fix line weight

### CODE REVIEW

**Comment by fantahund** [comment]
does not work with Optifabric/Optifine. No idea if this is due to Optifine/Optifabrig or to WorldeditCUI

**Comment by james090500** [comment]
> does not work with Optifabric/Optifine. No idea if this is due to Optifine/Optifabrig or to WorldeditCUI

OptiFabric isn't updated to 1.18.1 yet?

**Comment by cs127** [comment]
awesome

**Comment by Drift91** [comment]
> > does not work with Optifabric/Optifine. No idea if this is due to Optifine/Optifabrig or to WorldeditCUI
> 
> OptiFabric isn't updated to 1.18.1 yet?
> 
> -- Edit
> 
> It literally just came out, i'll do some testing

Any progress? I'm fairly certain this is relevant:
https://github.com/EngineHub/WorldEditCUI/pull/34#issuecomment-888031565
Prior to the commit linked in the above comment the compatibility was fixed.

**Comment by IARI** [comment]
I have built and tested, and you know that already, but I can confirm:
- without optifabric it works
- with optifabric, nothing is drawn.

**Comment by james090500** [comment]
The biggest issue I currently face is unable to get OptiFabric to load in my dev environment. [#523](https://github.com/Chocohead/OptiFabric/issues/523)

**Comment by octylFractal** [comment]
Official stance on the matter -- we should not delay a release for Optifine support. If someone wishes to fix it later, they may PR it themselves, until then I would say it is not something that time should be spent fixing.

Before this is merged, I and @wizjany think the line width issue should be resolved, if it can be done in a timely manner.

**Comment by zml2008** [comment]
optifine integration is likely fairly trivial to fix - just a matter of making sure the right render sink is chosen. it shouldn't block the release (plus a lot of people want it for some reason) 

line weight is trickier, but really just needs someone who understands the math to sort things out

**Comment by Drift91** [comment]
> (plus a lot of people want it for some reason)

I honestly wouldn't be surprised if more than half the people who want to use WorldEditCUI also use OptiFabric/Optifine.

**Comment by IARI** [comment]
Maybe it was just me, but the reason I wanted it in the first place was because simply I haden't even heard about any alternatives.
Now I have Sodium running and WECUI works fine with it.

**Comment by Drift91** [comment]
> Maybe it was just me, but the reason I wanted it in the first place was because simply I haden't even heard about any alternatives. Now I have Iris+Sodium running and WECUI works fine with it.

I tried them out in 1.17 and had issues. I might try them again. Overall though Optifine has a lot of features in a single, simple package. If I recall correctly, when I tried out Iris and Sodium there were a select few features that weren't covered that without, Minecraft pretty much makes my eyes bleed. I guess that's just me though since I'm fairly picky about things like mipmapping and anisotropic filtering.

File: build.gradle.kts
Comment (zml2008) [inline build.gradle.kts:21]: no, let me know what is needed from here and i'll publish it, rather than using this garbage.

File: build.gradle.kts
Comment (zml2008) [inline build.gradle.kts:50]: add parchment in here for parameter names

File: src/main/java/com/mumfrey/worldeditcui/event/listeners/CUIRenderContext.java
Comment (zml2008) [inline src/main/java/com/mumfrey/worldeditcui/event/listeners/CUIRenderContext.java:26]: rename this method to align with the offirial name

File: src/main/java/com/mumfrey/worldeditcui/render/CUIRenderLayers.java
Comment (zml2008) [inline src/main/java/com/mumfrey/worldeditcui/render/CUIRenderLayers.java:7]: fix this name

File: src/main/java/eu/mikroskeem/worldeditcui/mixins/MinecraftClientAccess.java
Comment (zml2008) [inline src/main/java/eu/mikroskeem/worldeditcui/mixins/MinecraftClientAccess.java:12]: rename

File: src/main/java/eu/mikroskeem/worldeditcui/render/IrisPipelineProvider.java
Comment (zml2008) [inline src/main/java/eu/mikroskeem/worldeditcui/render/IrisPipelineProvider.java:14]: why was this deleted? is it no longer necessary?

File: src/main/java/eu/mikroskeem/worldeditcui/render/VanillaPipelineProvider.java
Comment (zml2008) [inline src/main/java/eu/mikroskeem/worldeditcui/render/VanillaPipelineProvider.java:17]: yikes no

File: build.gradle.kts
Comment (zml2008) [inline build.gradle.kts:59]: why did you remove this? it helps with mixin inspection

**Review by zml2008 — CHANGES_REQUESTED** [review]

File: build.gradle.kts
Comment (james090500) [reply build.gradle.kts:21]: OptiFabric. Depends if we are going to support it or not, otherwise it wont be needed.

File: src/main/java/eu/mikroskeem/worldeditcui/render/VanillaPipelineProvider.java
Comment (james090500) [reply src/main/java/eu/mikroskeem/worldeditcui/render/VanillaPipelineProvider.java:17]: This is from #34 I will remove.

File: src/main/java/eu/mikroskeem/worldeditcui/render/IrisPipelineProvider.java
Comment (james090500) [reply src/main/java/eu/mikroskeem/worldeditcui/render/IrisPipelineProvider.java:14]: I spoke with Iris and we confirmed it wasn't needed anymore.

**Comment by james090500** [comment]
The Line width is somewhat fixed. Except the cube now seems to vibrate...

I will be away from my PC for the next 2 weeks so will look when back

https://user-images.githubusercontent.com/20372645/147287262-667048d7-1a11-434d-af99-a57dc46fca5b.mp4

**Review by zml2008 — CHANGES_REQUESTED** [review]
good to see that progress! it's likely some issue with the math behind LINES_LOOP handling, or perhaps something with normals calculation?

File: build.gradle.kts
Comment (james090500) [reply build.gradle.kts:21]: we will do this

File: src/main/java/com/mumfrey/worldeditcui/event/CUIEventType.java
Comment (james090500) [reply src/main/java/com/mumfrey/worldeditcui/event/CUIEventType.java:3]: Yeh intelij seems to like doing that itself.

File: src/main/java/eu/mikroskeem/worldeditcui/render/VanillaPipelineProvider.java
Comment (adryd325) [inline src/main/java/eu/mikroskeem/worldeditcui/render/VanillaPipelineProvider.java:16]:
```suggestion
        private static final BufferBuilderRenderSink.RenderType QUADS = new BufferBuilderRenderSink.RenderType(VertexFormat.Mode.QUADS, DefaultVertexFormat.POSITION_COLOR, GameRenderer::getPositionColorShader);
```

File: src/main/java/eu/mikroskeem/worldeditcui/render/VanillaPipelineProvider.java
Comment (adryd325) [inline src/main/java/eu/mikroskeem/worldeditcui/render/VanillaPipelineProvider.java:17]:
```suggestion
        private static final BufferBuilderRenderSink.RenderType LINES = new BufferBuilderRenderSink.RenderType(VertexFormat.Mode.LINES, DefaultVertexFormat.POSITION_COLOR_NORMAL, GameRenderer::getRendertypeLinesShader);
```

File: src/main/java/eu/mikroskeem/worldeditcui/render/VanillaPipelineProvider.java
Comment (adryd325) [inline src/main/java/eu/mikroskeem/worldeditcui/render/VanillaPipelineProvider.java:18]:
```suggestion
        private static final BufferBuilderRenderSink.RenderType LINES_LOOP = new BufferBuilderRenderSink.RenderType(VertexFormat.Mode.LINES, DefaultVertexFormat.POSITION_COLOR_NORMAL, GameRenderer::getRendertypeLinesShader);
```

**Review by adryd325 — CHANGES_REQUESTED** [review]
These changes fix the lines flickering issue

image

File: src/main/java/eu/mikroskeem/worldeditcui/render/VanillaPipelineProvider.java
Comment (adryd325) [reply src/main/java/eu/mikroskeem/worldeditcui/render/VanillaPipelineProvider.java:18]: not sure how to have made it all one suggestion, sorry >.<

**Comment by james090500** [comment]
This is now looking beautifully stable. The last bug is the rendering behind clouds but ONLY when "Fabulous" graphics are enabled....

image

File: src/main/java/eu/mikroskeem/worldeditcui/render/VanillaPipelineProvider.java
Comment (james090500) [reply src/main/java/eu/mikroskeem/worldeditcui/render/VanillaPipelineProvider.java:16]: What different does this make?

Comment (adryd325) [reply src/main/java/eu/mikroskeem/worldeditcui/render/VanillaPipelineProvider.java:16]: its the way quads should be rendered, though it doesnt seem to make any difference in game

Comment (james090500) [reply src/main/java/eu/mikroskeem/worldeditcui/render/VanillaPipelineProvider.java]: I have revered back, if you check. Let me know if that's right.

Comment (adryd325) [reply src/main/java/eu/mikroskeem/worldeditcui/render/VanillaPipelineProvider.java]: looks good

**Comment by james090500** [comment]
This seems to be good to go!

image

**Comment by zml2008** [comment]
I've been experimenting with this a little bit -- unfortunately I'm pretty sure the clouds thing is just a tradeoff we're going to have to accept in order to both a) be compatible with rendering mods and b) properly handle hidden line renders, though that can be fiddled with after the initial release.

Additionally @james090500, could I ask you to comment on #56 and confirm that you're willing to contribute to WorldEditCUI under the terms of the GPL v3 or later? We've just begun our relicensing effort, and your support is appreciated.

**Pull request merged**
